### PR TITLE
Fix benchmarks and adds CI to test them

### PR DESCRIPTION
**Pull request by bkchr**
Instead of waiting for benchmarks failing in Polkadot CI, we also can just test them in Substrate :P

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]

File: scripts/ci/gitlab/pipeline/test.yml [hunk 1]
```diff
@@ -277,6 +277,26 @@ test-linux-stable-extra:
...
+    - .test-refs
+  variables:
+    # Enable debug assertions since we are running optimized builds for testing
+    # but still want to have debug assertions.
+    RUSTFLAGS:                     "-Cdebug-assertions=y -Dwarnings"
+    RUST_BACKTRACE:                1
+    WASM_BUILD_NO_COLOR:           1
+    WASM_BUILD_RUSTFLAGS:          "-Cdebug-assertions=y -Dwarnings"
+    # Ensure we run the UI tests.
+    RUN_UI_TESTS:                  1
```
Comment (dvdplm) [inline scripts/ci/gitlab/pipeline/test.yml:294]: Are there UI tests among the benchmarks?

File: scripts/ci/gitlab/pipeline/test.yml [hunk 2]
```diff
@@ -277,6 +277,26 @@ test-linux-stable-extra:
     - time cargo test --doc --workspace --locked --release --verbose --features runtime-benchmarks --manifest-path ./bin/node/cli/Cargo.toml
     - rusty-cachier cache upload
 
+# This job runs all benchmarks defined in the `/bin/node/runtime` once to check that there are no errors.
+short-benchmark:
```
Comment (dvdplm) [inline scripts/ci/gitlab/pipeline/test.yml:281]: "short" here sounds a bit weird? Maybe `node-benchmarks`, `partial-benchmarks` or `quick-benchmarks`?

**Review by dvdplm — APPROVED** [review]

File: scripts/ci/gitlab/pipeline/test.yml (hunk 2 above)
Comment (bkchr) [reply scripts/ci/gitlab/pipeline/test.yml:281]: I borrowed the name from the Polkadot side :D

**Review by sergejparity — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]
Thanks!

**Review by rcny — APPROVED** [review]

**Pull request merged**